chanbackup/pubsub.go

```diff
@@ -168,6 +184,39 @@ func (s *SubSwapper) Stop() error {
 	return nil
 }
 
+// ManualUpdate inserts/updates channel states into the swapper. The updates
+// are processed in another goroutine. The method waits for the updates to be
+// fully processed and the file to be updated on-disk before returning.
+func (s *SubSwapper) ManualUpdate(singles []Single) error {
+	// Create the update object to insert into the processing loop.
+	errChan := make(chan error, 1)
+	update := manualUpdate{
+		singles: singles,
+		errChan: errChan,
+	}
+
+	select {
```

Comment:
In order to do it, we need `singles`. In case of ManualUpdate, they are passed by the caller. If we add time based updates, we need to run the timer outside of `SubSwapper`, because it does not have the source of this information (`server.chanStateDB` and `server.addrSource`).

Reply:
ahh ok not super trivial gottya